dao/src/main/java/com/epam/esm/dao/entity/Tag.java

```diff
@@ -18,6 +54,14 @@ public void setName(String name) {
     this.name = name;
   }
 
+  public Set<GiftCertificate> getGiftCertificates() {
+    return (giftCertificates == null) ? null : new HashSet<>(giftCertificates);
```

Comment:
@TimurRadko Why do we need this logic here?

Reply:
Because I return a copy of Set, and in the case, when it is equal null, we get NullPointerException. This case can happen when during the creation copy of the collection we pass to constructor arg==null.